src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs

```diff
@@ -0,0 +1,144 @@
+
+        [Activate]
+        [HtmlAttributeNotBound]
+        public IJavaScriptStringEncoder JavaScriptEncoder { get; set; }
+
+         /// <inheritdoc />
+        public override void Process(TagHelperContext context, TagHelperOutput output)
+        {
+            // Pass through attribute that is also a well-known HTML attribute.
+            if (Src != null)
+            {
+                output.CopyHtmlAttribute(SrcAttributeName, context);
+            }
+            
+            // NOTE: Values in TagHelperOutput.Attributes may already be HTML-encoded.
+            var attributes = new TagHelperAttributeList(output.Attributes);
+            var builder = new DefaultTagHelperContent();
+
+            if (context.AllAttributes.ContainsName(FileVersionAttributeName))
+            {
+                // Build a <img /> tag to match the original one in the source file.
+                // Or just add file version to the img tag.
+                BuildImageTag(attributes, builder);
+            } else
+            {
+                //No attributes matched so we have nothing to do
+                return;
+            }
+
+            // We've taken over tag rendering, so prevent rendering the outer tag
+            output.TagName = null;
+            output.Content.SetContent(builder);
+        }
+        
+        private void EnsureFileVersionProvider()
+        {
+            if (_fileVersionProvider == null)
+            {
+                _fileVersionProvider = new FileVersionProvider(
+                    HostingEnvironment.WebRootFileProvider,
+                    Cache,
+                    ViewContext.HttpContext.Request.PathBase);
+            }
+        }
+
+        private void BuildImageTag(TagHelperAttributeList attributes, TagHelperContent builder)
+        {
+            EnsureFileVersionProvider();
+            builder.Append("<img ");
+
+            foreach (var attribute in attributes)
+            {
+                var attributeValue = attribute.Value;
+                if (FileVersion == true &&
+                    string.Equals(attribute.Name, SrcAttributeName, StringComparison.OrdinalIgnoreCase))
+                {
+                    // "src" values come from bound attributes and globbing. So anything but a non-null string is
+                    // unexpected but could happen if another helper targeting the same element does something odd.
+                    // Pass through existing value in that case.
+                    var attributeStringValue = attributeValue as string;
+                    if (attributeStringValue != null)
+                    {
+                        attributeValue = _fileVersionProvider.AddFileVersionToPath(attributeStringValue);
+                    }
+                }
+
+                builder
+                    .Append(attribute.Name)
+                    .Append("=\"")
+                    .Append(HtmlEncoder, ViewContext.Writer.Encoding, attributeValue)
```

Comment:
For every attribute _except_ `Src`, this usually double-encode the attribute value.